llsd/serde_xml.py

```diff
@@ -143,59 +178,18 @@ class LLSDXMLPrettyFormatter(LLSDXMLFormatter):
     This class is not necessarily suited for serializing very large objects.
     It sorts on dict (llsd map) keys alphabetically to ease human reading.
     """
-    def __init__(self, indent_atom = None):
+    def __init__(self, indent_atom = b'  '):
         "Construct a pretty serializer."
         # Call the super class constructor so that we have the type map
         super(LLSDXMLPrettyFormatter, self).__init__()
 
         # Private data used for indentation.
-        self._indent_level = 1
-        if indent_atom is None:
-            self._indent_atom = b'  '
-        else:
-            self._indent_atom = indent_atom
+        self._indent_atom = indent_atom
```

Comment:
This is how we get around the assignment at line 76.

But - since the base class now has a no-op `_indent()` method, does the base class even need `_indent_atom`? Does anything else in `LLSDXMLFormatter` reference that attribute?

Maybe replace the `LLSDXMLFormatter.__init__(indent_atom)` argument with an `eol` argument, and set the base-class `_eol` attribute from that? `LLSDXMLPrettyFormatter.__init__()` could accept both `indent_atom` and `eol`.

Reply:
restructuring.